server/src/test/java/org/spine3/server/storage/ProjectionStorageShould.java

```diff
@@ -69,6 +85,208 @@ public void return_null_if_no_event_time_in_storage() {
+                                                                  .setVersion(1)
+                                                                  .build();
+            storage.write(id, record);
+            ids.add(id);
+        }
+
+        final Map<I, EntityStorageRecord> read = storage.readAll();
+        assertSize(ids.size(), read);
+        for (Map.Entry<I, EntityStorageRecord> record : read.entrySet()) {
+            assertContains(record.getKey(), ids);
+        }
+    }
+
+    @SuppressWarnings("MethodWithMultipleLoops")
+    @Test
+    public void read_all_messages_with_field_mask() {
+        final int count = 5;
+        final List<I> ids = new LinkedList<>();
+
+        final String projectDescriptor = Project.getDescriptor()
+                                                .getFullName();
+        @SuppressWarnings("DuplicateStringLiteralInspection")
+        final FieldMask fieldMask = FieldMask.newBuilder()
+                                             .addPaths(projectDescriptor + ".id")
+                                             .addPaths(projectDescriptor + ".name")
+                                             .build();
+
+        for (int i = 0; i < count; i++) {
+            final I id = newId();
+            final ProjectId stateId = ProjectId.newBuilder()
+                                               .setId(id.toString())
+                                               .build();
+            final Project state = Project.newBuilder()
+                                         .setId(stateId)
+                                         .setName(String.format("project_%s", i))
+                                         .setStatus(Project.Status.CREATED)
+                                         .build();
+            final Any packedState = AnyPacker.pack(state);
+
+            final EntityStorageRecord record = EntityStorageRecord.newBuilder()
+                                                                  .setState(packedState)
+                                                                  .setWhenModified(Timestamps.getCurrentTime())
+                                                                  .setVersion(1)
+                                                                  .build();
+            storage.write(id, record);
+            ids.add(id);
+        }
+
+        final Map<I, EntityStorageRecord> read = storage.readAll(fieldMask);
+        assertSize(ids.size(), read);
+        for (Map.Entry<I, EntityStorageRecord> record : read.entrySet()) {
+            assertContains(record.getKey(), ids);
+
+            final Any packedState = record.getValue()
+                                          .getState();
+            final Project state = AnyPacker.unpack(packedState);
+            assertMatchesMask(state, fieldMask);
+        }
+    }
+
+    @Test
+    public void retrieve_empty_map_if_store_is_empty() {
```

Comment:
Is it `store` or `storage`? Looking at the rest of the tests in this class I would suggest to rename the method.